src/runtime/TemporalObject.cpp

```diff
@@ -513,15 +504,92 @@ TemporalObject::DateTime TemporalObject::parseTemporalInstantString(ExecutionSta
 {
     auto result = parseValidIso8601String(state, isoString, true);
 
-    if (!result.z.empty()) {
-        result.offsetString = "+00:00";
+    if (result.tz->z) {
+        result.tz->offsetString = new ASCIIString("+00:00");
     }
 
-    ASSERT(!result.offsetString.empty());
+    if (!result.tz->offsetString || result.tz->offsetString->length() == 0) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "undefined offsetString");
+    }
+
+    return result;
+}
+
+TemporalObject::TimeZone TemporalObject::parseTemporalTimeZoneString(ExecutionState& state, const std::string& isoString)
+{
+    auto result = TimeZone(false, String::emptyString, String::emptyString);
+
+    if (isoString == "UTC") {
+        result.z = true;
+        return result;
+    }
+
+    unsigned int index = 0;
+    result = TemporalObject::parseTimeZoneOffset(state, const_cast<std::string&>(isoString), index);
+
+    if (result.offsetString->length() == 0 && result.name->length() == 0 && !result.z) {
+        result = *TemporalObject::parseValidIso8601String(state, isoString, true).tz;
+    }
+
+    if (result.z) {
+        result.offsetString = String::emptyString;
+    }
+
+    if (!(result.offsetString && result.name) && result.z) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "bare date-time string is not a time zone");
+    }
 
     return result;
 }
 
+TemporalObject::TimeZone TemporalObject::parseTimeZoneOffset(ExecutionState& state, std::string& isoString, unsigned int& index)
+{
+    // Offset
+    auto tz = TemporalObject::TimeZone{ false, String::emptyString, String::emptyString };
```

Comment:
`TimeZone(false, String::emptyString, String::emptyString);` used above. Is this needs to be different?